waspc/src/Generator/Job/Process.hs

```diff
@@ -42,8 +46,40 @@ runProcessAsJob process jobType = \chan -> do
     return exitCode
 
 runNodeCommandAsJob :: Path Abs (Dir a) -> String -> [String] -> J.JobType -> J.Job
-runNodeCommandAsJob fromDir command args jobType = do
-    -- TODO: Check npm/node version.
-    let process = (P.proc command args) { P.cwd = Just $ SP.toFilePath fromDir }
-    runProcessAsJob process jobType
+runNodeCommandAsJob fromDir command args jobType chan = do
+    errorOrNodeVersion <- getNodeVersion
+    case errorOrNodeVersion of
+        Left errorMsg -> exitWithError (ExitFailure 1) errorMsg
+        Right nodeVersion -> if nodeVersion < C.nodeVersion
+            then exitWithError (ExitFailure 1)
+                 ("Your node version is too low, it should be >= " ++ C.nodeVersionAsText)
+            else do
+                let process = (P.proc command args) { P.cwd = Just $ SP.toFilePath fromDir }
+                runProcessAsJob process jobType chan
+  where
+      exitWithError exitCode errorMsg = do
+          writeChan chan $ J.JobMessage
+              { J._data = J.JobOutput errorMsg J.Stderr
+              , J._jobType = jobType }
+          writeChan chan $ J.JobMessage { J._data = J.JobExit exitCode
+                                        , J._jobType = jobType }
+          return exitCode
 
+      getNodeVersion :: IO (Either String (Int, Int, Int))
+      getNodeVersion = do
+          (exitCode, stdout, stderr) <- P.readProcessWithExitCode "node" ["--version"] ""
+          return $ case exitCode of
+              ExitFailure _ -> Left $ "Running 'node --version' failed: " ++ stderr
+              ExitSuccess -> case parseNodeVersion stdout of
+                                 Nothing -> Left "Wasp failed to parse node version."
+                                 Just version -> Right version
+
+      parseNodeVersion :: String -> Maybe (Int, Int, Int)
+      parseNodeVersion nodeVersionStr =
+          case nodeVersionStr R.=~ ("v([^\\.]+).([^\\.]+).(.+)" :: String) of
```

Comment:
maybe extract this bad boy regex to `where` below, and give it some nice name? but ok